athenz/resource_role.go

```diff
@@ -117,9 +147,23 @@ func ResourceRole() *schema.Resource {
 				},
 			},
 		},
+		CustomizeDiff: validateRoleSchema,
 	}
 }
 
+func validateRoleSchema(ctx context.Context, d *schema.ResourceDiff, meta interface{}) error {
+	_, mNew := d.GetChange("member")
+	members := mNew.(*schema.Set).List()
+
+	_, sNew := d.GetChange("settings")
+	settings := map[string]interface{}{}
+	if len(sNew.(*schema.Set).List()) > 0 {
+		settings = sNew.(*schema.Set).List()[0].(map[string]interface{})
+	}
+
+	return validateRoleMember(members, settings)
```

Comment:
please add some null checks.
we don't want to get segmentation faults, but present a nice error to the user

Reply:
I tested null cases, but this won't happen here because if `mNew` or `sNew` are nil, the conversion of `.(*schema.Set).List()` converts to an empty list. It's also tested in one of the tests in resource_role_test: TestAccRoleSettingsMemberExpiryAndReview.